app/util/notifications/hooks/index.test.ts

```diff
@@ -0,0 +1,249 @@
+                id: '123',
+              },
+            },
+          },
+        }),
+      );
+      await waitFor(() => {
+        expect(notifee.onForegroundEvent).toHaveBeenCalled();
+      });
+    });
+  });
+
+  it('should do nothing if the EventType is DISMISSED', async () => {
+    const { waitFor } = renderHook(() =>
+      useNotificationHandler(mockNavigation),
+    );
+
+    await act(async () => {
+      notifee.onForegroundEvent(() =>
+        mockNotificationEvent({
+          type: EventType.DISMISSED,
+          detail: {
+            notification: {
+              body: 'notificationTest',
+              data: {
+                action: 'tx',
+                id: '123',
+              },
+            },
+          },
+        }),
+      );
+
+      await waitFor(() => {
+        expect(notifee.onForegroundEvent).toHaveBeenCalled();
+      });
+    });
+  });
+
+  it('should do nothing if data.action is not tx', async () => {
+    const { waitFor } = renderHook(() =>
+      useNotificationHandler(mockNavigation),
+    );
+
+    await act(async () => {
+      notifee.onForegroundEvent(() =>
+        mockNotificationEvent({
+          type: EventType.DELIVERED,
+          detail: {
+            notification: {
+              body: 'notificationTest',
+              data: {
+                action: 'no-tx',
+                id: '123',
+              },
+            },
+          },
+        }),
+      );
+
+      await waitFor(() => {
+        expect(notifee.onForegroundEvent).toHaveBeenCalled();
+      });
+
+      expect(mockNavigate).not.toHaveBeenCalled();
+    });
+  });
+
+  it('handleOpenedNotification should do nothing if notification is null', async () => {
+    const { waitFor } = renderHook(() =>
+      useNotificationHandler(mockNavigation),
+    );
+
+    await act(async () => {
+      notifee.onForegroundEvent(() =>
+        mockNotificationEvent({
+          type: EventType.DELIVERED,
+          detail: {
+            notification: undefined,
+          },
+        }),
+      );
+      await waitFor(() => {
+        expect(notifee.onForegroundEvent).toHaveBeenCalled();
+      });
+
+      expect(mockNavigate).not.toHaveBeenCalled();
+    });
+  });
+
+  it('should navigate to the transaction view when the notification action is "tx"', async () => {
+    const { waitFor } = renderHook(() =>
+      useNotificationHandler(mockNavigation),
+    );
+
+    await act(async () => {
+      notifee.onForegroundEvent(() =>
+        mockNotificationEvent({
+          type: EventType.DELIVERED,
+          detail: {
+            notification: {
+              body: 'notificationTest',
+              data: {
+                action: 'tx',
+                id: '123',
+              },
+            },
+          },
+        }),
+      );
+      await waitFor(() => {
+        expect(notifee.onForegroundEvent).toHaveBeenCalled();
+      });
+    });
+  }, 10000);
+
+  it('should process notification on Android', async () => {
```

Comment:
```suggestion
  it('processes notification on Android', async () => {
```
Q: why this Android specific test case?

Reply:
Wrongly added since we're testing it on the `displays notifications correctly`